thunder/tests/test_inplace_functionalization.py

```diff
@@ -476,31 +476,27 @@ def f(xs, ys, z):
     dtypes=NOTHING,
 )
 def test_inplace_to_tensors_with_grad(executor, device, _):
-    @torch.no_grad
```

Comment:
I am not sure what exactly do we test for here and also why did we get rid of `add_grad` case.

Reply:
The test came with the addition of the `grad` member variable for `TensorProxy`.  The `add_grad` case is removed because I was having trouble making the test make sense for even pytorch operations and I don't see what it adds on top of the `add_y` case.

Reply:
Actually I think this test should still work as is.

PyTorch allows update on leaf tensors under `no_grad`.
```python
import torch

@torch.no_grad  # Commenting will lead to error
def add_y(x, y):
    x.add_(y, alpha=0.1)

x = torch.randn(3, 3, requires_grad=True)
y = torch.randn(3, 3, requires_grad=True)

add_y(x, y)
```

Also, I think `add_grad` is important as it was added in the PR that added support for `grad` attribute on TensorProxy: https://github.com/Lightning-AI/lightning-thunder/pull/1070, so it seems to be testing that as well. (We should probably add comment that this is happening).

Reply:
OK, test restored, and, I dare say, even improved.